test/helpers.ts

```diff
@@ -107,9 +107,14 @@ export function runQuery(
   assertFn?: (res: any[]) => void,
   log = true,
   withUserAndPassword = () => ({ username: config.db.user, password: config.db.password }),
+  queryReplacements: () => { [key: string]: string } = () => ({}),
```

Comment:
What is this, and why can't it be done with normal `` ` `` strings?

Reply:
no, jest does like a first pass grabbing all statements and the current variable values. I guess it is because you can run tests in parallel and tests should be independent? Since we are assigning these values in another test assert step, we have to do this trickery to have the latest value for the variable. It replicates the same behaviour as `username` and `password`.

Reply:
oh, that can be resolved by wrapping in an extra `async (done) => ...` closure scope. We do that in a few places already.

Reply:
I tried something like that too, but it was not working when calling the `query` method inside since it has its own logic returning a (done) => void function itself

Reply:
It's ugly, but [this is the pattern we use](https://github.com/iasql/iasql/blob/main/test/modules/ssh-apt-integration.ts#L204-L223)